onnxruntime/core/session/plugin_ep/ep_kernel_registration.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <memory>
+#include "core/session/onnxruntime_c_api.h"
+#include "core/framework/data_types.h"
+#include "core/framework/error_code_helper.h"
+#include "core/framework/kernel_def_builder.h"
+#include "core/framework/kernel_registry.h"
+#include "core/framework/op_kernel.h"
+
+struct OrtMLDataType : onnxruntime::DataTypeImpl {};
+
+struct OrtKernelDefBuilder : onnxruntime::KernelDefBuilder {};
+
+struct OrtKernelDef : onnxruntime::KernelDef {};
```

Comment:
I wonder if these should inherit from the `onnxruntime::` types? do we ever instantiate one of these derived types? if the pointers always refer to an `onnxruntime::` type instance, maybe we can just use pointers to opaque types that we cast to the relevant `onnxruntime::` type as needed.